docs/jwst/extract_2d/reference_files.rst

```diff
@@ -1,26 +1,91 @@
-optional correction (on by default). It can be turned off by specifying
-``apply_wavecorr=False`` when running the step.
+The WAVECORR reference file contains pixel offset values as a function of
+wavelength and source offset within a NIRSpec slit.
+It is used when applying the NIRSpec wavelength zero-point correction to
+fixed-slit (EXP_TYPE="NRS_FIXEDSLIT"), bright object TSO
+(EXP_TYPE="NRS_BRIGHTOBJ"), and MSA/MOS spectra (EXP_TYPE="NRS_MSASPEC").
+This is an optional correction that is turned on by default.
+It can be turned off by specifying ``apply_wavecorr=False`` when running the step.
 
 .. include:: wavecorr_selection.rst
 
-WAVELENGTHRANGE Reference
--------------------------
+.. include:: ../includes/standard_keywords.rst
+
+Type Specific Keywords for WAVECORR
++++++++++++++++++++++++++++++++++++++
+In addition to the standard reference file keywords listed above,
+the following keywords are *required* in WAVECORR reference files,
+because they are used as CRDS selectors
+(see :ref:`wavecorr_selectors`):
+
+=========  ==============================
+Keyword    Data Model Name
+=========  ==============================
+EXP_TYPE   model.meta.exposure.type
+=========  ==============================
+
+Reference File Format
++++++++++++++++++++++
+WAVECORR reference files are ASDF format, with the format and contents
+specified by the `WaveCorrModel` data model schema.
+
+WAVELENGTHRANGE Reference File
+------------------------------
 
-NIRCAM WFSS and NIRISS WFSS observations use the wavelengthrange reference file
-in order to construct the bounding boxes around each objects orders. If a list
-of ``GrismObject`` is supplied, then no reference file is neccessary.
+:REFTYPE: WAVELENGTHRANGE
+:Data model: `WavelengthrangeModel`
+
+The WAVELENGTHRANGE reference file contains information on the minimum and
+maximum wavelengths of various spectroscopic modes, which can be
+order-dependent. The reference data are used to construct bounding boxes
+around the spectral traces produced by each object in the NIRCam and NIRISS
+WFSS modes.
+If a list of ``GrismObject`` is supplied, then no reference file is neccessary.
 
 .. include:: wavelengthrange_selection.rst
 
+.. include:: ../includes/standard_keywords.rst
+
+Type Specific Keywords for WAVELENGTHRANGE
+++++++++++++++++++++++++++++++++++++++++++
+In addition to the standard reference file keywords listed above,
+the following keywords are *required* in WAVELENGTHRANGE reference files,
+because they are used as CRDS selectors
+(see :ref:`wavelengthrange_selectors`):
+
+=========  ==============================
+Keyword    Data Model Name
+=========  ==============================
+EXP_TYPE   model.meta.exposure.type
+=========  ==============================
+
+Reference File Format
++++++++++++++++++++++
+WAVELENGTHRANGE reference files are ASDF format, with the format and contents
+specified by the `WavelengthragneModel` data model schema.
```

Comment:
typo: Wavelengthra**gn**eModel

BTW, single back ticks will attempt to link the type. If this doesn't work it needs to be specified as `~jwst.datamodels.WavelengthrangeModel` so that it links correctly.

Reply:
Ooh, I was wondering if it was somehow possible to include a link to each data model. Now I know how. Thanks!